src/scripts/genXplatEventing.py

```diff
@@ -339,6 +338,12 @@ def generateClrallEvents(eventNodes,allTemplates):
         fnptype.extend(fnptypeline)
         fnptype.append("\n)\n{\n")
         fnbody.append(lindent)
+        fnbody.append("if (!EventEnabled")
```

Comment:
I don't think you need this call to EventEnabled<EventName>.  Here is some example generated code:

inline ULONG FireEtwGCStart(
                  const unsigned int  Count,
                  const unsigned int  Reason
)
{
                  if (!EventEnabledGCStart()) {return ERROR_SUCCESS;}
                  EventPipeWriteEventGCStart(Count,Reason);
                  return FireEtXplatGCStart(Count,Reason);
}

The call to EventEnabled is duplicate logic because the EventPipe write and the LTTng write will both do an IsEnabled check for the specific logging mechanism.  You should be able to safely remove this.  We still want the EventEnabled<Event> functions generated so that they can be called by VM code, they just don't need to be called here.